scripts/Create-AdoBuildAgent.ps1

```diff
@@ -75,40 +75,42 @@ Write-Debug "${functionName}:subnetId=$subnetId"
-        $output = az login --service-principal -u $env:servicePrincipalId -p $env:servicePrincipalKey --tenant $imageGalleryTenantId
-        Write-Debug ([array]$output | Out-String)
-        $output = az account get-access-token
+        $command = "az login --service-principal -u $($env:servicePrincipalId) -p $($env:servicePrincipalKey) --tenant $imageGalleryTenantId"
+        $commandOutput = Invoke-CommandLine -Command $command -ReturnExitCode
+        $command = "az account get-access-token"
+        $commandOutput = Invoke-CommandLine -Command $command -ReturnExitCode
     }
 
-    $output = az login --service-principal -u $env:servicePrincipalId -p $env:servicePrincipalKey --tenant $tenantId
-    Write-Debug ([array]$output | Out-String)
-    $output = az account get-access-token
+    $command = "az login --service-principal -u $($env:servicePrincipalId) -p $($env:servicePrincipalKey) --tenant $tenantId"
+    $commandOutput = Invoke-CommandLine -Command $command -ReturnExitCode
+    $command = "az account get-access-token"
+    $commandOutput = Invoke-CommandLine -Command $command -ReturnExitCode
+
+    $command = "az account set --subscription $subscriptionName"
+    $commandOutput = Invoke-CommandLine -Command $command -ReturnExitCode
 
-    $output = az account set --subscription $subscriptionName
-    Write-Debug ([array]$output | Out-String)
+    Write-Host "Checking if the VMSS $vmssName already exists..."
+    $command = "az vmss list --resource-group $resourceGroup"
+    $commandOutput = Invoke-CommandLine -Command $command
 
-    $instances = az vmss list --resource-group $resourceGroup | ConvertFrom-Json
+    $instances = $commandOutput | ConvertFrom-Json
     if (-not ($instances.name -contains $vmssName)) {
         Write-Host "Creating VMSS: $vmssName..."
-        $output = az vmss create `
-            --resource-group $resourceGroup `
-            --name $vmssName `
-            --computer-name-prefix $vmssName `
-            --vm-sku Standard_D4s_v4 `
-            --instance-count 2 `
-            --subnet $subnetId `
-            --image "$imageId" `
-            --authentication-type password `
-            --admin-username $adoAgentUser `
-            --admin-password "$adoAgentPass" `
-            --disable-overprovision `
-            --upgrade-policy-mode Manual `
-            --public-ip-address '""' `
-            --tags ServiceName='ADP' ServiceCode='CDO' Name=$vmssName Purpose='ADO Build Agent'
-        Write-Debug ([array]$output | Out-String)
+        
+        $command = "az vmss create --resource-group $resourceGroup --name $vmssName --computer-name-prefix $vmssName --vm-sku Standard_D4s_v4 --instance-count 2 --subnet '$subnetId' --image '$imageId' --authentication-type password --admin-username $adoAgentUser --admin-password '$adoAgentPass' --disable-overprovision --upgrade-policy-mode Manual --public-ip-address '""""'  --tags ServiceName='ADP' ServiceCode='CDO' Name=$vmssName Purpose='ADO Build Agent'"
```

Comment:
Can we re-format this as above? clearer to read

Reply:
I'll check again... tried to keep it multi-line as before but because of the Invoke-Command function usage had to change it to a single line string.

Reply:
And can we update the VMSS settings after creation? i.e. sku, instance, policies, tags etc?

Reply:
That's not scripted but we can change the SKU, policies, and tags if needed.
The instance counts are managed by ADO once the pool is configured to use the scale-set.

Reply:
how do we change the policies? just update the parameters in effect or something more elaborate ?

Reply:
This script will not do it as this just creates the instances.. we will have to change it manually or extend the script to support updates.

Reply:
We need to extend it ideally ... manually doesnt exist 

maybe in next story....